spec/fixtures/mock_upstream.lua

```diff
@@ -1,12 +1,94 @@
-local utils = require "kong.tools.utils"
+local utils      = require "kong.tools.utils"
+local cjson_safe = require "cjson.safe"
+local cjson      = require "cjson"
+local ws_server  = require "resty.websocket.server"
+
+
+local function string_trim(s)
+  local _,i1 = s:find("^%s*")
+  return s:sub(i1 + 1, s:find("%s*$") - 1)
+end
+
+
+local function string_beginswith(s, prefix)
+  return s:sub(1, #prefix) == prefix
+end
+
+
+local function parse_multipart_form_params(body, content_type)
```

Comment:
Several reasons. First is that I didn't realize that we had lua-multipart as a dependency already :/. I used it as a base, but I changed some things because in order to make the tests pass I need less information (I didn't need the parameter order or its headers) and in a different format (I needed a dictionary-like table, not a hibrid dictionary-array one). I think the code for using lua-multipart and then removing the information I don't need and transforming would be similar in size to just parsing the thing.